akka-stream-tests/src/test/scala/akka/stream/io/StreamTcpSpec.scala

```diff
@@ -375,48 +390,42 @@ class StreamTcpSpec extends AkkaSpec with TcpHelper {
           .via(echoConnection)
           .runFold(ByteString.empty)((acc, in) ⇒ acc ++ in)
 
-      Await.result(resultFuture, 5.seconds) should be(expectedOutput)
-      Await.result(binding.unbind(echoServerMM), 3.seconds)
+      Await.result(resultFuture, 3.seconds) should be(expectedOutput)
+      Await.result(binding.unbind(), 3.seconds)
       Await.result(echoServerFinish, 1.second)
     }
 
     "bind and unbind correctly" in {
       val address = temporaryServerAddress()
-      val binding = StreamTcp(system).bind(address)
       val probe1 = StreamTestKit.SubscriberProbe[StreamTcp.IncomingConnection]()
-      val mm1 = binding.connections.to(Sink(probe1)).run()
-      probe1.expectSubscription()
-
+      val bind = StreamTcp(system).bind(address)
       // Bind succeeded, we have a local address
-      Await.result(binding.localAddress(mm1), 1.second)
+      val binding1 = Await.result(bind.to(Sink(probe1)).run(), 3.second)
+
+      probe1.expectSubscription()
 
       val probe2 = StreamTestKit.SubscriberProbe[StreamTcp.IncomingConnection]()
-      val mm2 = binding.connections.to(Sink(probe2)).run()
+      val binding2F = bind.to(Sink(probe2)).run()
       probe2.expectErrorOrSubscriptionFollowedByError(BindFailedException)
 
       val probe3 = StreamTestKit.SubscriberProbe[StreamTcp.IncomingConnection]()
-      val mm3 = binding.connections.to(Sink(probe3)).run()
+      val binding3F = bind.to(Sink(probe3)).run()
       probe3.expectErrorOrSubscriptionFollowedByError()
 
-      // The unbind should NOT fail even though the bind failed.
-      Await.result(binding.unbind(mm2), 1.second)
-      Await.result(binding.unbind(mm3), 1.second)
-
-      an[BindFailedException] shouldBe thrownBy { Await.result(binding.localAddress(mm2), 1.second) }
-      an[BindFailedException] shouldBe thrownBy { Await.result(binding.localAddress(mm3), 1.second) }
+      an[BindFailedException] shouldBe thrownBy { Await.result(binding2F, 1.second) }
```

Comment:
unbinding changed now in that it makes only sense once you got a ServerBinding. So these semantic changes are valid IMHO